src/openrct2/Game.h

```diff
@@ -18,6 +18,124 @@
 class Intent;
 struct ParkLoadResult;
 
+// TODO: Move or remove these constants if
+//       we can use the constants defined
+//       in Context.h
+namespace TickConversion
+{
+    constexpr uint32_t kGameUpdateFPS = 40;
+    constexpr float kGameUpdateFPMS =  kGameUpdateFPS / 1000.0f;
+} // namespace TickConversion
+
+template <typename T, typename Tag>
+struct GameTicks
+{
+    using ValueType = T;
+
+    ValueType Value;
+
+    static constexpr GameTicks FromSeconds(uint32_t seconds)
+    {
+        GameTicks<T, Tag> result;
+        result.Value = seconds * TickConversion::kGameUpdateFPS;
+        return result;
+    }
+    static constexpr GameTicks FromMilliseconds(uint32_t milliseconds)
+    {
+        GameTicks<T, Tag> result;
+        result.Value = milliseconds * TickConversion::kGameUpdateFPMS;
```

Comment:
We don't need to use floating point math here, see my remark on `kGameUpdateFPMS`. We should also round it because 1 millisecond would result to 0 otherwise, in this case we have to represent a single tick starting at 0 ms to 25 ms. The formula should be something like:

```suggestion
        result.Value = (milliseconds + (TickConversion::kGameUpdateTimeMs - 1)) / TickConversion::kGameUpdateTimeMs /* 25 */;
```